server.go

```diff
@@ -74,6 +75,10 @@ type server struct {
 	// long-term identity private key.
 	lightningID [32]byte
 
+	natTraversal    nat.Traversal
+	lastIP          net.IP
+	advertisedPorts []uint16
```

Comment:
can `advertisedPorts` be moved inside the `natTraversal` instead? That way you can pass it a list of ports when creating it, and get a list of complete addresses when calling `ExternalIP`. With the option to automatically find ports.

Reply:
Fixed.